server/src/instant/db/model/attr.clj

```diff
@@ -257,6 +257,25 @@
                      "Namespaces are not allowed to start with a `$`.
                       Those are reserved for system namespaces.")}])))))
 
+(defn validate-add-required! [conn app-id attrs]
+  (doseq [attr  attrs
+          :when (:required? attr)
+          :let  [[_ etype label] (:forward-identity attr)
+                 query "SELECT *
+                          FROM idents
+                          JOIN attrs ON attrs.forward_ident = idents.id
+                          JOIN triples ON triples.attr_id = attrs.id
+                         WHERE idents.etype = ?etype
+                           AND idents.app_id = ?app-id
+                           AND triples.app_id = ?app-id
+                         LIMIT 1"
+                 res   (sql/execute! conn (sql/format query {"?etype" etype, "?app-id" app-id}))]
+          :when (seq res)]
+    (ex/throw-validation-err!
+     :attributes
+     attr
+     [{:message (str "Can't set attribute `" label "` as required because `" etype "` already have entities")}])))
```

Comment:
nit: perhaps we say something like 'can't create new attribute ` label

(set could imply this is an update, when really we are creating)